core/shared/src/main/scala/scalaz/zio/Queue.scala

```diff
@@ -16,60 +16,26 @@
 
 package scalaz.zio
 
-import scala.annotation.tailrec
 import scalaz.zio.internal.Platform
-import scalaz.zio.Queue.internal._
+import scalaz.zio.Queue2.internal._
 import scalaz.zio.internal.MutableConcurrentQueue
 
+import scala.annotation.tailrec
+
 /**
- *  A `Queue[A]` is a lightweight, asynchronous queue for values of type `A`.
+ * A `Queue2[R, E, A, B]` is a lightweight, asynchronous queue into which values of type `A`
```

Comment:
You still have the 4 types here instead of the 6.

Reply:
Fixed